cubids/cubids.py

```diff
@@ -366,19 +458,22 @@ def change_filename(self, filepath, entities):
 
         Parameters
         ----------
-        filepath : str
-            Path prefix to a file in the affected key group change
-        entities : dictionary
-            A pybids dictionary of entities parsed from the new key
-            group name.
+        filepath : :obj:`str`
+            Path prefix to a file in the affected key group change.
+        entities : :obj:`dict`
+            A pybids dictionary of entities parsed from the new key group name.
+
+        Notes
+        -----
+        This is the function I need to spend the most time on, since it has entities hardcoded.
         """
         exts = Path(filepath).suffixes
-        old_ext = ""
-        for ext in exts:
-            old_ext += ext
+        old_ext = "".join(exts)
 
         suffix = entities["suffix"]
         entity_file_keys = []
+
+        # Entities that may be in the filename?
         file_keys = ["task", "acquisition", "direction", "reconstruction", "run"]
 
         for key in file_keys:
```

Comment:
@mattcieslak do you know what these entities are hardcoded here for? Are they the only entities that are used to group files? For example, if two files are the same but have different contrast agent (`ce`) entities, would CuBIDS treat them as part of the same key group?

Reply:
I don't think we had ever seen ce entities at this point. These were from what we'd seen while curating RBC and should definitely be expanded to better match current bids

Reply:
Gotcha. Does that mean I was interpreting the code correctly then? I.e., that those are the entities that are used to group files?